### PR TITLE
refactors the way we compare lists in unit tests

### LINKED ISSUE #2000: validating HTTP responses that contains an array of items in tests
**User Story** Some HTTP endpoints return responses that contain an array of items. The response is not stable in a way that items are actually not sorted. To validate if the response is correct we have to cast the response to a strongly typed slice and double loop when comparing with desired list.

It is desirable to find a solution that would compare the response string with expected one not taking ordering of the items into account.

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: defines helper types that expose convenient methods
that are used by the test to sort the actual and expected list and then perform the comparison.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2000 

```release-note
NONE
```

### CODE REVIEW

File: api/pkg/handler/project_test.go [hunk 1]
```diff
@@ -119,39 +125,13 @@ func TestListProjectEndpoint(t *testing.T) {
...
-				rawBody, err := ioutil.ReadAll(res.Body)
-				if err != nil {
-					t.Fatal(err)
-				}
-
-				err = json.Unmarshal(rawBody, &projectsFromResponse)
-				if err != nil {
-					t.Fatal(err)
-				}
+			actualProjects := ProjectV1SliceWrapper{}
```
Comment (p0lyn0mial) [inline api/pkg/handler/project_test.go]: that the essence of the changes, now the flow is
```
actualProjects := ProjectV1SliceWrapper{}
actualProjects.DecodeOrDie(res.Body, t).Sort()

wrappedExpectedProjects := ProjectV1SliceWrapper(tc.ExpectedResponse)
wrappedExpectedProjects.Sort()

actualProjects.EqualOrDie(wrappedExpectedProjects, t
```

**Review by kdomanski — DISMISSED** [review]
I actually think this makes the code less and not more clear, but hey, you own the code. ;-) There's just a typo in `defaultCreatoinTimestamp`.

**Comment by p0lyn0mial** [comment]
> I actually think this makes the code less and not more clear

@glower What do you think ? Should we remove the wrappers and move the logic to the tests ?

asking one more person before we will make the final decision. I'm fine with reverting this pull if that what we want.

**Review by glower — DISMISSED** [review]
I think this solution is OK to fix the problem with our tests now

**Comment by p0lyn0mial** [comment]
I fixed the typo in `defaultCreatoinTimestamp` and made the new types to be not exported. Can one of you tag the pull ?

**Review by kdomanski — APPROVED** [review]

**Pull request merged**